### PR TITLE
Setup build and test harness for rules as js files (#145)

### PR DESCRIPTION
**Pull request by shawnmclean**
- [x] CI script added
- [x] Jest test harness added
- [x] Build script implemented for building `rules.json` from plain js files using jsdoc for metadata.

Rules:
- [x] access-on-weekdays-only-for-an-app
- [x] encrypt-sensitive-data
- [x] decrypt-sensitive-data
- [x] fraud-prevention-with-minfraud
- [x] mixpanel-track-event

### CODE REVIEW

File: build-new [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    alias: 'a',
+    default: false
+  })
+  .option('outputPath', {
+    alias: 'o',
+    default: __dirname + '/rules.json'
+  })
+  .argv;
+
+var files = fs.readdirSync(__dirname + '/src/rules').map(function (file) {
```
Comment (fyockm) [inline build-new]: `var` vs `const`?

File: build-new [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  //group by category
+  var rules = results;
+
+  var categories = rules.filter(function (rt) {
+    return !!rt;
+  }).map(function (rule) {
+    return rule.categories;
+  }).reduce(function (prev, curr) {
+    var new_cats = curr.filter(function (cat) {
+      return !~prev.indexOf(cat);
```
Comment (fyockm) [inline build-new]: while I understand what you're doing here, I think the bitwise operation could be confusing for other devs unfamiliar with this JS "trick". Wish we could just use [`includes`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes), but need to support Node 4 for the time being.

File: test/utils/load-rule.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+function compile(code, globals, stubs) {
+  function fakeRequire (moduleName) {
+    if (stubs[moduleName]) {
+      return stubs[moduleName];
+    }
+
+    return require(moduleName);
+  }
+
+  const globalObj = Object.assign({}, { require: fakeRequire }, globals);
```
Comment (fyockm) [inline test/utils/load-rule.js]: https://github.com/auth0/auth0-authz-rules-api/blob/master/index.js might be useful in helping define the "magic globals" that are used in Rules.

File: build-new [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+}, function (err, results) {
+  if (err) {
+    console.log(err.stack);
+    process.exit(1);
+  }
+
+  //group by category
+  var rules = results;
+
+  var categories = rules.filter(function (rt) {
```
Comment (fyockm) [inline build-new]: seems like this map => filter => reduce (with filter) => map could be simplified

File: test/utils/load-rule.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+  const file = fs.readFileSync(fileName, 'utf8');
+
+  const code =  decomment(file).trim();
+
+  return compile(code, globals, stubs);
+}
+
+function compile(code, globals, stubs) {
+  function fakeRequire (moduleName) {
```
Comment (fyockm) [inline test/utils/load-rule.js]: more simply `return stubs[moduleName] || require(moduleName)`

File: build-new (hunk 2 above)
Comment (shawnmclean) [reply build-new]: `build-new` was copied and modified from `build` (https://github.com/auth0/rules/blob/master/build). I'm unfamiliar with the trick, but I avoided touching code below line's 45.

File: build-new [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    callback(null, result);
+  });
+}, function (err, results) {
+  if (err) {
+    console.log(err.stack);
+    process.exit(1);
+  }
+
+  //group by category
+  const rules = results;
```
Comment (fyockm) [inline build-new]: use of a modern `async` will also bring along `lodash` at no extra charge. so, might as well leverage [_.groupBy](https://lodash.com/docs/4.17.5#groupBy) here.

File: test/rules/access-on-weekdays-only-for-an-app.test.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+  beforeEach(() => {
+    globals = {
+      UnauthorizedError: function() {}
+    };
+    context = {
+      clientName: 'TheAppToCheckAccessTo'
+    };
+
+    rule = loadRule('access-on-weekdays-only-for-an-app.js', globals);
```
Comment (fyockm) [inline test/rules/access-on-weekdays-only-for-an-app.test.js]: let's just set a `const` at the top of test with the `testName`. Then you can use that as the `describe` message.

**Pull request merged**